### DISCUSSION TITLE
Docs are incorrect regarding UploadFile

### DISCUSSION THREAD

**Discussion by holyoaks** [fastapi/fastapi#9922] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import UploadFile

file_name = 'test.png'
file_path = 'test.png'

with open(file_path, "rb") as f:
    file=UploadFile(
        filename=file_name, file=f.read(), content_type="image/png"
    )
)
```

### Description

1. Docs [here](https://fastapi.tiangolo.com/tutorial/request-files/?h=uploadfile#uploadfile) specify UploadFile has a `content_type` attribute, which it used to
2. Running the code above with any file throws `TypeError: UploadFile.__init__() got an unexpected keyword argument 'content_type'`
3. Actual Starlette code (datastructures.py) does not expect `content_type`:
```
def __init__(
        self,
        file: typing.BinaryIO,
        *,
        size: typing.Optional[int] = None,
        filename: typing.Optional[str] = None,
        headers: "typing.Optional[Headers]" = None,
    )
```

Docs need to be updated to accurately reflect changes to Starlette

### Operating System

Linux

### FastAPI Version

0.100.0

### Python Version

3.11

**Comment by mateoradman** [comment, marked as answer]
Hi @holyoaks,

I believe what is stated in the docs is absolutely correct. That is, an instance of an UploadFile indeed has a `content_type` attribute. Having an attribute does not imply that UploadFile accepts content_type in the `__init__` method.

Looking into the signature of the UploadFile class one can see that a content_type is a property method. This means that this attribute content_type is calculate on the fly every time you access `some_file.content_type` attribute.

```python
class UploadFile:
    """
    An uploaded file included as part of the request data.
    """

    def __init__(
        self,
        file: typing.BinaryIO,
        *,
        size: typing.Optional[int] = None,
        filename: typing.Optional[str] = None,
        headers: "typing.Optional[Headers]" = None,
    ) -> None:
        self.filename = filename
        self.file = file
        self.size = size
        self.headers = headers or Headers()

    @property
    def content_type(self) -> typing.Optional[str]:
        return self.headers.get("content-type", None)
```

**Reply by holyoaks** [reply]
Hmm, fair enough. It was an unexpected change, as previously the Starlette code _did_ accept `content_type`.
